### ISSUE TITLE
error when trying to force download

### ISSUE THREAD

**Issue by D3nn3** [MrBrax/TwitchAutomator#10]
Hi mate!
first, thanks for the app.

I got it running via docker and the app starts, I can update the settings and add streamers, so far so good. But it seems when a stream start the app does not record it and when I try to force record I get the following error:

```
Slim Application Error
The application could not run because of the following error:

Details
Type: Error
Code: 0
Message: Call to a member function getTimestamp() on bool
File: /var/www/twitchautomator/src/TwitchVOD.php
Line: 991
Trace
#0 /var/www/twitchautomator/src/TwitchVOD.php(192): App\TwitchVOD->parseChapters(Array)
#1 /var/www/twitchautomator/src/TwitchVOD.php(255): App\TwitchVOD->load('/var/www/twitch...')
#2 /var/www/twitchautomator/src/TwitchAutomator.php(578): App\TwitchVOD->refreshJSON()
#3 /var/www/twitchautomator/src/TwitchAutomator.php(365): App\TwitchAutomator->capture(Array)
#4 /var/www/twitchautomator/src/TwitchAutomator.php(205): App\TwitchAutomator->download(Array)
#5 /var/www/twitchautomator/src/Routes/routes.php(80): App\TwitchAutomator->handle(Array)
#6 /var/www/twitchautomator/vendor/slim/slim/Slim/Handlers/Strategies/RequestResponse.php(43): Closure->{closure}(Object(Slim\Psr7\Request), Object(Slim\Psr7\Response), Array)
#7 /var/www/twitchautomator/vendor/slim/slim/Slim/Routing/Route.php(381): Slim\Handlers\Strategies\RequestResponse->__invoke(Object(Closure), Object(Slim\Psr7\Request), Object(Slim\Psr7\Response), Array)
#8 /var/www/twitchautomator/vendor/slim/slim/Slim/MiddlewareDispatcher.php(81): Slim\Routing\Route->handle(Object(Slim\Psr7\Request))
#9 /var/www/twitchautomator/vendor/slim/slim/Slim/MiddlewareDispatcher.php(81): Slim\MiddlewareDispatcher->handle(Object(Slim\Psr7\Request))
#10 /var/www/twitchautomator/vendor/slim/slim/Slim/Routing/Route.php(341): Slim\MiddlewareDispatcher->handle(Object(Slim\Psr7\Request))
#11 /var/www/twitchautomator/vendor/slim/slim/Slim/Routing/RouteRunner.php(84): Slim\Routing\Route->run(Object(Slim\Psr7\Request))
#12 /var/www/twitchautomator/vendor/slim/slim/Slim/Middleware/RoutingMiddleware.php(60): Slim\Routing\RouteRunner->handle(Object(Slim\Psr7\Request))
#13 /var/www/twitchautomator/vendor/slim/slim/Slim/MiddlewareDispatcher.php(140): Slim\Middleware\RoutingMiddleware->process(Object(Slim\Psr7\Request), Object(Slim\Routing\RouteRunner))
#14 /var/www/twitchautomator/vendor/slim/twig-view/src/TwigMiddleware.php(125): class@anonymous->handle(Object(Slim\Psr7\Request))
#15 /var/www/twitchautomator/vendor/slim/slim/Slim/MiddlewareDispatcher.php(140): Slim\Views\TwigMiddleware->process(Object(Slim\Psr7\Request), Object(class@anonymous))
#16 /var/www/twitchautomator/vendor/slim/slim/Slim/Middleware/ErrorMiddleware.php(107): class@anonymous->handle(Object(Slim\Psr7\Request))
#17 /var/www/twitchautomator/vendor/slim/slim/Slim/MiddlewareDispatcher.php(140): Slim\Middleware\ErrorMiddleware->process(Object(Slim\Psr7\Request), Object(class@anonymous))
#18 /var/www/twitchautomator/vendor/slim/slim/Slim/MiddlewareDispatcher.php(81): class@anonymous->handle(Object(Slim\Psr7\Request))
#19 /var/www/twitchautomator/vendor/slim/slim/Slim/App.php(215): Slim\MiddlewareDispatcher->handle(Object(Slim\Psr7\Request))
#20 /var/www/twitchautomator/vendor/slim/slim/Slim/App.php(199): Slim\App->handle(Object(Slim\Psr7\Request))
#21 /var/www/twitchautomator/public/index.php(136): Slim\App->run()
#22 {main}
```

Only after stopping the container and deleting the created json file ind the folder of the streamer the dashboard works again. Any help would be much appreciated!

edit: It just caught my eye that the image on dockerhub doesn't seem to be up2date (last updated 1 month ago even though this repo has commits to it 2 days ago). Maybe this has already been fixed and updating the image would be enough. Also using semantic versioning in addition to the latest tag would be nice on dockerhub :)

**Comment by MrBrax** [comment]
yeah pushing docker images was just too much for me, you should be able to build the image manually though. but you're gonna have to add cronjobs manually too, since i couldn't find a solution for that either. basically, docker is just annoying for this

**Comment by D3nn3** [comment]
Hi, thanks for answering so fast.

As for building and pushing docker images: It's super easy, you just need 3 commands in your terminal.

```
docker login --username=<dockerhub username>
docker build -t mrbrax/twitchautomator:latest -f Dockerfile .
docker push mrbrax/twitchautomator:latest
```

Since most people will run this on a NAS or something similar and nowadays everyone just runs docker images because it doesn't get more comfortable than that.
As for the cron jobs: Can't really help you with that but there are enough docker images of applications around that do scheduled tasks. Don't know how they do it exactly but it can be done.

**Comment by MrBrax** [comment]
i've checked out docker a bit more, it seems that the alpine linux image doesn't have php 7.4 yet, so i'll have to wait for that

**Comment by D3nn3** [comment]
Hi, sorry for the late reply.

Looked at the Dockerfile: You could just use alpine as the base image and install nginx + php 7.4. I'm sure there are other Dockerfiles around where people do exactly that. Could just look how they do it but it shouldn't really that complicated.

**Comment by MrBrax** [comment]
Fixed in 5f80469d53964a8d4df733bd5dafd4875bf2e537
